### ISSUE TITLE
Cookie-Disclaimer at login page

### ISSUE THREAD

**Issue by usox** [ampache/ampache#2771]
Due to some technical problems and the actual requirement to inform about cookies before actually setting them, we will have to inform about the cookie-situation on the login page.

Ensure:
- Display the cookie disclaimer only if activated in the settings
- Show the disclaimer once and remember the consent (by using a cookie...)

**Comment by lachlan-00** [comment]
Does something like this (notification and implied consent from logging into the site) meet gdpr requirements?

Ampache doesn't have anything outside required site functionality so i think this would cover it?
![image](https://user-images.githubusercontent.com/1305249/108641907-a341c600-74ed-11eb-9acd-a2b51c465714.png)

Do we also create a cookie upon log in to disable the message?

**Comment by wagnered** [comment]
A clause could indicate "By logging in you agree to the use of cookies."   Lately. I've noticed that almost all have the text "Got it?" On the close button  of the pop-up cookie notice. Thus no choice is given.

**Comment by lachlan-00** [comment]
haha got it is a bit snarky.
I guess we should also allow a custom message for this warning too

**Comment by lachlan-00** [comment]
you're right about the "by logging in" part though that's the consent action

**Comment by lachlan-00** [comment]
probably link to a 'policy' page as well
https://github.com/ampache/ampache/wiki/cookies

Then include a link to the policy and blurb the text

```
Ampache places cookies on your computer to help make this website better.
Cookies are used for core site functionality and are not used for tracking or analytics.
By logging in you agree to the use of cookies while using this site. [policy link]
```

is the removal of this message needed when it's on the login page? i don't really see it as needed unless it's required that it be hidden

**Comment by wagnered** [comment]
> is the removal of this message needed when it's on the login page? i don't really see it as needed unless it's required that it be hidden

Considering that at login it is not known if the user is registered or not, it should always be displayed if the admin so desires to enable the cookie notice.  I don't think it matters if the message is "removed" or hidden.

**Comment by lachlan-00** [comment]
added in fb6c1c92a

I would backport to master but none of the new strings would be translated yet so i can move as part of 5

**Comment by usox** [comment]
> Does something like this (notification and implied consent from logging into the site) meet gdpr requirements?
> 
> Ampache doesn't have anything outside required site functionality so i think this would cover it?

Looking good.
 
> Do we also create a cookie upon log in to disable the message?

Well, would be nice (the old noty-warning did this actually).

**Comment by AshotN** [comment]
I think it should be hidden by default. Ampache is something you usually running for yourself, and if it is a service that is being hosted for others you can then enable it.

**Comment by lachlan-00** [comment]
You have to enable the disclaimer in config (US rules by default) and I think it would also be fine to set a cookie to hide it after first logon (which I haven't done yet either)

**Comment by AshotN** [comment]
Will the React Client need this if I am only using a cookie to store session?

**Comment by usox** [comment]
@AshotN In theory, law says you don't need a consent for technical required cookies (sessions, shopping baskets, ...)  but you should inform about the usage - especially in commercial usage scenarios (for the lawyer-bots: This is NO legal advise).
With the upcoming e-privacy law in EU, "privacy by design" is going to be enforced. Meaning: The provided defaults should ensure a maximum of privacy which can be lowered by the user itself.
